cpp/src/sender.cpp

```diff
@@ -64,10 +64,17 @@ namespace {
 uint64_t tag_counter = 0;
 }
 
-tracker sender::send(const message &message) {
+tracker sender::send(const message &message,const binary &tag) {
+    pn_delivery_t *dlv;
+    if(!tag.empty())
+    {
+    dlv = pn_delivery(pn_object(), pn_dtag((std::string(tag)).c_str(), tag.size()));
```

Comment:
This could be done in a nicer way. `binary` is a vector; you are guaranteed that the individual values will be in memory one after another, so even something like `(char *) &tag[0]` should work, no need to convert to string, then do c_str.

Reply:
(but use a c++-style cast, I did not do it above because I don't know which one is appropriate)

Reply:
Done. Thanks!